cli/pcluster/config/param_types.py

```diff
@@ -434,6 +434,31 @@ def get_cfn_value(self):
 # that require some custom action during CFN/file conversion
 
 
+class ExtraJsonParam(JsonParam):
+    """Class to manage extra_json configuration parameters."""
+
+    def to_cfn(self):
+        """
+        Convert param to CFN representation.
+
+        The extra_json configuration parameter can contain both "cfncluster" or "cluster" keys but cookbook
+        recipes require "cfncluster" as key.
+        """
+        if self.value and "cluster" in self.value:
```

Comment:
I looked back to the code and I think the best way to replace the `self.value` during CFN conversion is by overriding the `get_cfn_value` and not the `to_cfn` method.

This means that we can avoid to override the `to_cfn` and simply override the:
```python
    def get_cfn_value(self):
        """
        Convert parameter value into CFN value.

        The extra_json configuration parameter can contain both "cfncluster" or "cluster" keys but cookbook
        recipes require "cfncluster" as key.
       """
       self.value = {"cfncluster": self.value.pop("cluster")}
       return self.get_string_value()
```

then it will be automatically called by the `to_cfn` of the `Param` class (see: https://github.com/aws/aws-parallelcluster/blob/develop/cli/pcluster/config/param_types.py#L185).

I think it is the best way because I don't like that in the `get_cfn_value` we are emitting the parameter not sanitized, and by overriding this method we'll have the right sanitization in both `to_cfn` and `get_cfn_value` outputs.
The unit test below must be updated accordingly to it, see my following comment.